src/components/ThemeMenu/Modal/ThemeModal.tsx

```diff
@@ -0,0 +1,246 @@
+ *
+ * Leveraging the ThemeContext to apply the theme changes across the application.
+ * User selection is dispatched to ThemeContext, which handles the theme update and storing.
+ *
+ * @returns A modal that allows the choice of a theme and renders a preview of different app pages based on the selection
+ * @category Components
+ */
+
+const ThemeModal = ({ open = false, handleClose}: ThemeModalProps) => {
+
+    const { t } = useTranslation()
+
+    //gets theme from user and provides string
+    const activeTheme = useTheme()
+    const activeThemeString = activeTheme === DarkTheme ? 'DarkTheme':
+        activeTheme === AltTheme ? 'AltTheme':
+            'HaskiTheme';
+
+    //presets first selection on currently active theme
+    const [selectedTheme, setSelectedTheme] = useState(activeTheme)
+    const [selectedThemeString, setSelectedThemeString] = useState(activeThemeString)
+
+    const {updateTheme} = useThemeContext()
+    const {isAuth} = useAuthProvider()
+    const isSmallScreen = useMediaQuery(activeTheme.breakpoints.down('lg'));
+
+    //handles the selection of a radio button
+    const handleThemeModalPreviewChange= (themeName: string) => {
+        setSelectedTheme(themeName === 'DarkTheme'? DarkTheme :
+            themeName ===  'AltTheme'? AltTheme :
+                HaskiTheme
+        );
+    }
+
+    //Will preset the user stored theme as the one shown in preview and selected on user auth
+    useEffect(() => {
+        setSelectedTheme(activeTheme)
+        setSelectedThemeString(activeThemeString)
+    }, [ isAuth]);
+
+    //PreviewPageChangerLogic
+    const [pageIndex, setPageIndex] = useState(0)
+    const pages = [
+        <ThemePresentation key="theme"/>,
+        <AboutUs key="about"/>,
+        <PrivacyPolicy key="privacy"/>,
+        <Course key="course"/>,
+        <Topic key="topic"/>,
+        <Home key=""/>
+    ]
+
+    //will iterate through the pages on use of arrow buttons
+    const changePage = (direction : number) => {
+        setPageIndex((prevIndex) =>
+            (prevIndex + direction + pages.length) % pages.length)
+    }
+
+    return (
+        <Modal data-testid={'Theme Modal'} open={open} onClose={handleClose}>
+            <Box sx={styleBox}>
+                {/**Left column/Upper Row - Provides radio buttons for theme changing*/}
+                <Box sx={leftColumnStyle}>
+                    <FormControl sx={leftColumnInner}>
+                        <FormLabel id="demo-radio-buttons-group-label">{t('components.ThemeModal.radioHeader')}</FormLabel>
+                        <RadioGroup
+                            sx={{
+                                flexDirection: { xs: 'row', lg: 'column' },
+                                gap: 2
+                            }}
+                            value={selectedThemeString}
+                            onChange={(e) => {setSelectedThemeString(e.target.value); handleThemeModalPreviewChange(e.target.value)}}
```

Comment:
setSelectedThemeString könnte man in handleThemeModal... moven, da es immer zusammen ausgeführt wird und auch dieselbe value hat